pkg/endpoint/bpf.go

```diff
@@ -712,6 +712,20 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, boo
 		}
 	}
 
+	// Walk the L4Policy to add new redirects and update the desired policy map
+	// state to set the newly allocated proxy ports.
+	if e.DesiredL4Policy != nil {
+		desiredRedirects, err = e.addNewRedirects(owner, e.DesiredL4Policy)
```

Comment:
I think this commit should be reverted. With this, we're calling `addNewRedirects` twice. Here, and on line 700 above. I will send a PR to revert.

Reply:
See https://github.com/cilium/cilium/pull/5481.